src/ucs/memory/memtype_cache.h

```diff
@@ -116,6 +102,18 @@ void ucs_memtype_cache_remove(ucs_memtype_cache_t *memtype_cache,
  */
 void ucs_memory_info_set_host(ucs_memory_info_t *mem_info);
 
+
+/**
+ * Find if global memtype_cache is empty.
+ *
+ * @return 1 if empty 0 if otherwise.
+ */
+static UCS_F_ALWAYS_INLINE int ucs_memtype_cache_is_empty()
+{
+    return ((ucs_memtype_cache_global_instance != NULL) &&
+            !ucs_memtype_cache_global_instance->pgtable.num_regions);
```

Comment:
(ucs_memtype_cache_global_instance->pgtable.num_regions == 0)

Reply:
Use of memtype_cache_is_empty is not guaranteed to check if global instance is null or not. If I make the suggested change, it  leads to a segfault if memtype_cache is disabled.